core/src/main/scala/com/yahoo/maha/core/query/QueryGenerator.scala

```diff
@@ -404,6 +414,22 @@ object QueryGeneratorHelper {
     }
   }
 
+  def overrideRenderedColWithTimezone(
+                           newTimeZone: Option[String]
+                           , column: DerivedColumn
+                           , query: String
+                         ): String = {
+    column match {
+      case HiveDerDimCol(_, dt, _, de, _, _, _) =>
+        de.expression match {
+          case TIME_FORMAT_WITH_TIMEZONE(_, fmt, tz) if newTimeZone.isDefined && tz != newTimeZone.get =>
+            query.replaceAll(tz, newTimeZone.get)
```

Comment:
can we move this at the query generation instead of using string literal replace all ?

Reply:
Hi @pranavbhole , thanks for reviewing the PR :) Do you mean re-render the derived expression with the new timezone value? 
Currently, the rendering is happening while registering the table, when a request coming, we are using the rendered expression to generate the query directly according to this https://github.com/yahoo/maha/blob/master/core/src/main/scala/com/yahoo/maha/core/DerivedExpression.scala#L957. 
And for this timezone case, we would like to apply the specified timezone which defined in the additionalParameter in the request, but if make this change by re-rendering the expression, it will affect other requests who are running in parallel and want to use different timezones, since all these requests are sharing the same register facts info

Reply:
Hi @pranavbhole, updated the PR which instead of using the replaceAll(), it is using an internal function in TIME_FORMAT_WITH_TIMEZONE to dynamic render is when generating query for a request, pls check